src/validation.rs

```diff
@@ -0,0 +1,158 @@
+        }
+    }
+
+    fn invalid_message(&self) -> &'static str {
+        match self {
+            ValidationField::Npub { id } => match id.as_str() {
+                "npub_buyer" => "Invalid buyer npub format. Please enter a valid Nostr public key.",
+                "npub_seller" => {
+                    "Invalid seller npub format. Please enter a valid Nostr public key."
+                }
+                "npub_arbitrator" => {
+                    "Invalid arbitrator npub format. Please enter a valid Nostr public key."
+                }
+                "npub_1" => "Invalid first npub format. Please enter a valid Nostr public key.",
+                "npub_2" => "Invalid second npub format. Please enter a valid Nostr public key.",
+                _ => "Invalid npub format. Please enter a valid Nostr public key.",
+            },
+            ValidationField::Amount { id } => match id.as_str() {
+                "amount_buyer" => "Buyer amount must be between 0.00000001 and 100 BTC.",
+                "amount_seller" => "Seller amount must be between 0.00000001 and 100 BTC.",
+                "amount_total" => "Total escrowed amount must be between 0.00000001 and 100 BTC.",
+                "amount" => "Resolution spend amount must be between 0.00000001 and 100 BTC.",
+                _ => "Amount must be between 0.00000001 and 100 BTC.",
+            },
+            ValidationField::FeeRate => "Fee rate must be a positive integer greater than zero.",
+            ValidationField::Nsec => "Invalid nsec format. Please enter a valid Nostr secret key.",
+            ValidationField::Txid => "Invalid transaction ID. Please enter a valid transaction ID.",
+            ValidationField::Transaction { id } => match id.as_str() {
+                "signed-tx" => "Invalid signed transaction format. Should be a hex string.",
+                "unsigned-tx" => "Invalid unsigned transaction format. Should be a hex string.",
+                _ => "Invalid transaction format. Should be a hex string.",
+            },
+            ValidationField::Signature { id } => match id.as_str() {
+                "signature1" => "Invalid first signature format.",
+                "signature2" => "Invalid second signature format.",
+                "signaturearb" => "Invalid arbitrator signature format.",
+                _ => "Invalid signature format.",
+            },
+            ValidationField::Address => {
+                "Invalid Bitcoin address format. Please check and try again."
+            }
+            ValidationField::EsploraUrl => {
+                "Invalid URL format. Should start with http:// or https://."
+            }
+            ValidationField::TimelockDays => "Days should be between 0 and 1,000.",
+            ValidationField::TimelockHours => "Hours should be between 0 and 23.",
+        }
+    }
+}
+
+pub(crate) fn validate_input(
+    input: &str,
+    field: ValidationField,
+    required: bool,
+) -> Option<String> {
```

Comment:
I find that a function named validate that returns `None` if the thing is validated somewhat unintuitive.